afterpay.php

```diff
@@ -0,0 +1,1340 @@
+    {
+        $currency = new Currency($cart->id_currency);
+
+        return array(
+            'button' => '#payment_button',
+            'currency_iso' => $currency->iso_code,
+            'cart_total' => $cart->getOrderTotal(),
+        );
+    }
+
+    /**
+     * Header hook
+     */
+    public function hookHeader()
+    {
+        echo '<!-- APVersion:'. $this->version.
+            ' PS:'._PS_VERSION_.
+            ' Env:'.Configuration::get('CLEARPAY_ENVIRONMENT').
+            ' MId:'.Configuration::get('CLEARPAY_PUBLIC_KEY').
+            ' Region:'.Configuration::get('CLEARPAY_REGION').
+            ' Lang:'.$this->getCurrentLanguage().
+            ' Enabled:'.Configuration::get('CLEARPAY_IS_ENABLED').
+            ' A_Countries:'.Configuration::get('CLEARPAY_ALLOWED_COUNTRIES').
+            ' R_Cat:'.(string)Configuration::get('CLEARPAY_RESTRICTED_CATEGORIES').
+            ' -->';
+        if (_PS_VERSION_ >= "1.7") {
+            $this->context->controller->registerJavascript(
+                sha1(mt_rand(1, 90000)),
+                self::AFTERPAY_JS_CDN_URL,
+                array('server' => 'remote')
+            );
+        } else {
+            $this->context->controller->addJS(self::AFTERPAY_JS_CDN_URL);
+        }
+    }
+
+    /**
+     * @return array
+     * @throws Exception
+     */
+    public function hookPaymentOptions()
+    {
+        /** @var Cart $cart */
+        $cart = $this->context->cart;
+        $this->shippingAddress = new Address($cart->id_address_delivery);
+        $this->billingAddress = new Address($cart->id_address_invoice);
+        $totalAmount = Afterpay::parseAmount($cart->getOrderTotal(true, Cart::BOTH));
+
+        $link = $this->context->link;
+
+        $return = array();
+        $this->context->smarty->assign($this->getButtonTemplateVars($cart));
+        $templateConfigs = array();
+        if ($this->isPaymentMethodAvailable()) {
+            $amountWithCurrency = Afterpay::parseAmount($totalAmount/4) . $this->currencySymbol;
+            if ($this->currency === 'GBP') {
+                $amountWithCurrency = $this->currencySymbol. Afterpay::parseAmount($totalAmount/4);
+            }
+            $checkoutText = $this->l('Or 4 interest-free payments of') . ' ' . $amountWithCurrency . ' ';
+            $checkoutText .= $this->l('with');
+            if ($this->isOPC()) {
+                $checkoutText = $this->l('4 interest-free payments of') . ' ' . $amountWithCurrency;
+            }
+            $templateConfigs['TITLE'] = (string) $checkoutText;
+            $language = Language::getLanguage($this->context->language->id);
+            if (isset($language['locale'])) {
+                $language = $language['locale'];
+            } else {
+                $language = $language['language_code'];
+            }
+
+            $templateConfigs['ISO_COUNTRY_CODE'] = str_replace('-', '_', $language);
+            if ($templateConfigs['ISO_COUNTRY_CODE'] == 'ca_ES') {
```

Comment:
This is not a valid language tag, it is very important to use valid tags in a global context.
```php
 public static function getFormattedLocaleTag()
    {
        $languageCode = str_replace('-','_',Context::getContext()->language->language_code);
        $languageTag = Tools::substr($languageCode, 0, 2);
        $geographicTag = Tools::strtoupper( Tools::substr($languageCode, 3, 5));
        $formattedLanguageTag =  $languageTag . '_'. $geographicTag;
        return $formattedLanguageTag;
    }
    `

Reply:
we can review it on the next release, right now is working properly and validated by QA Team